contracts/utils/FuseFixer.sol

```diff
@@ -0,0 +1,200 @@
+        address(0x5CaDc2a04921213DE60B237688776e0F1A7155E6), // Pool 156: FRAX
+        address(0x9CD060A4855290bf0c5aeD266aBe119FF3b01966), // Pool 156: DAI
+        address(0x74897C0061ADeec84D292e8900c7BDD00b3388e4), // Pool 156: LUSD
+        address(0x88d3557eB6280CC084cA36e425d6BC52d0A04429), // Pool 156: USDC
+        address(0xe92a3db67e4b6AC86114149F522644b34264f858) // Pool 156: ETH
+    ];
+
+    mapping(address => address[]) public underlyingsToCTokens;
+
+    constructor(address core) CoreRef(core) {
+        _buildCTokenMapping();
+
+        // check mappings lengths - hand calculated
+        assert(underlyingsToCTokens[UNDERLYINGS[0]].length == 7);
+        assert(underlyingsToCTokens[UNDERLYINGS[1]].length == 5);
+        assert(underlyingsToCTokens[UNDERLYINGS[2]].length == 5);
+        assert(underlyingsToCTokens[UNDERLYINGS[3]].length == 1);
+        assert(underlyingsToCTokens[UNDERLYINGS[4]].length == 5);
+        assert(underlyingsToCTokens[UNDERLYINGS[5]].length == 3);
+        assert(underlyingsToCTokens[UNDERLYINGS[6]].length == 2);
+        assert(underlyingsToCTokens[UNDERLYINGS[7]].length == 2);
+        assert(underlyingsToCTokens[UNDERLYINGS[8]].length == 4);
+        assert(underlyingsToCTokens[UNDERLYINGS[9]].length == 1);
+
+        // send out token approvals
+        _approveCTokens();
+    }
+
+    /// @dev Repay all debt
+    function repayAll() public onlyTribeRole(TribeRoles.PCV_SAFE_MOVER_ROLE) {
+        _repayETH();
+
+        // we skip index 0 because that's ETH
+        for (uint256 i = 1; i < UNDERLYINGS.length; i++) {
+            _repayERC20(UNDERLYINGS[i]);
+        }
+    }
+
+    /// @dev Repay the underlying asset on all ctokens up to the maximum provide
+    /// @notice reverts if the total bad debt is beyond the provided maximum
+    /// @param underlying the asset to repay in
+    /// @param maximum the maximum amount of underlying asset to repay
+    function repay(address underlying, uint256 maximum) public onlyTribeRole(TribeRoles.PCV_SAFE_MOVER_ROLE) {
+        require(getTotalDebt(underlying) < maximum, "Total debt is greater than maximum");
+
+        if (underlying == address(0)) {
+            _repayETH();
+        } else {
+            _repayERC20(underlying);
+        }
+    }
+
+    /// @dev Gets the total debt for the provided underlying asset
+    /// @notice This is not a view function! Use .staticcall in ethers to get the return value
+    /// @param underlying the asset to get the debt for; pass in 0x0 for ETH
+    /// @return debt the total debt for the asset
+    function getTotalDebt(address underlying) public returns (uint256 debt) {
+        for (uint256 i = 0; i < CTOKENS.length; i++) {
+            CTokenFuse token = CTokenFuse(CTOKENS[i]);
+            if (token.underlying() == underlying) {
+                debt += CTokenFuse(CTOKENS[i]).borrowBalanceCurrent(DEBTOR);
+            }
+        }
```

Comment:
couldn't you just iterate over the underlyingsToCtokens map ctokens rather than the whole CTokens. Then underlying is always =.

Reply:
That would be more gas efficient, yes.

Reply:
Leaving as-is since it works and this is a one-off.